app/controllers/application_controller.rb

```diff
@@ -1,2 +1,21 @@
 class ApplicationController < ActionController::Base
+	before_action :authenticate_user!
+	protect_from_forgery with: :exception
+	before_action :current_cart
+  
+	private
+	def current_cart
+		if session[:cart_id]            
+			cart = Cart.find_by(:id => session[:cart_id])          
```

Comment:
Correct indentions here. Use 2 spaces instead of 4

Reply:
done